Src/library/net/network/socket/network_socket.e

```diff
@@ -368,11 +439,11 @@ feature {NONE} -- Externals
 
 invariant
 
-	timeout_set: timeout > 0
+	timeout_set: timeout > 0 or (timeout_nano_seconds > 0 and timeout_nano_seconds < 1_000_000_000)
```

Comment:
Shouldn't there be assertions _timeout >= 0_ and _timeout_nano_seconds >= 0_?
Also, the current invariant does not allow 0 timeout, is it expected?